# bulk2sc_GMVAE
The `bulk2sc` model is a machine learning tool based on a Gaussian mixture variational autoencoder (GMVAE) that generates single-cell RNA sequencing (scRNA-seq) data from bulk RNA sequencing (bulk RNA-seq) data. The purpose of this repository is to demonstrate the operation of `bulk2sc` and to reproduce the results published in `TBA` for the scRNA-seq dataset [GSE130148](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE130148).
#### Fig.1 Model schematic diagram
<img src="fig/scRNAGMVAE.png" width="60%" alt="Image description">

The `bulk2sc` model is made up of three main components (**Fig.1**):
* I. scRNA Gaussian Mixture Variational Autoencoder (GMVAE) – Learns patterns in scRNA-seq data across different cell types and generates synthetic scRNA-seq data.
* II. Bulk RNA-seq Deconvolution – Predicts cell type proportions from bulk RNA-seq data using the training from scTAPE-seq data. The [`scTAPE`](https://sctape.readthedocs.io/) is adopted.
* III. scRNA-seq Generator – Integrates components I and II to generate scRNA-seq data from bulk RNA-seq input.

The GMVAE model (Part I in Fig. 1) requires a scRNA-seq raw count table in Matrix Market exchange format (MTX file) along with the corresponding cluster (cell type) data in a CSV file. The `scTAPE`(Part II in Fig. 1) utilizes the scRNA-seq raw count table as a dense matrix, with the cluster data as the row index. The column names are gene names. Additionally, TPM-normalized bulk data is needed to accompany the single-cell data.

Executing scripts sequentially from `_0_` to `_6_` will reproduce results in `TBA` for the scRNA-seq dataset [GSE130148](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE130148).

`_0_data_download` and `_1_data_process_for_GMVAE` prepare all necessary data for GMVAE training. The downloaded data is split into separate training and test datasets, each stored in dedicated folders. `_2_data_process_for_scTAPE` prepares the data for `scTAPE` training. The raw scRNA-seq training data is converted into a dense matrix format, and multiple pseudo-bulk RNA-seq datasets are generated by randomly sampling from the test data and applying TPM normalization. `_3_model_train_GMVAE` and `_4_model_train_scTAPE` train GMVAE and `scTAPE`, respectively, producing `.pt` files containing the saved models. These models are then used in `_5_scRNA_generation` to generate synthetic scRNA-seq data. This script generates scRNA-seq data for any specified cell-type-specific counts. Additionally, for comparison, it accepts scRNA-seq data, computes pseudo-bulk data, and returns a synthetic scRNA-seq dataset with a matching total number of cells. Finally, `_6_results_visualization` visualizes the generated data allowing comparison with ground truth data (if available). **Fig.2** illustrates the dependencies between these steps.

#### Fig.2 Procedures and operation order
<img src="fig/Figure2_operation_procedure_dependency2.png" width="45%" alt="Procedures and dependency">

<!---
#### Fig.3 scTAPE predicted vs true proportion. UMAPs of original test data vs generated data using predicted proportion.
<img src="fig/Figure3_results_umaps.png" width="50%" alt="Image description">
The trained models were applied to the test data from the lung dataset (GSE130148), which had not been seen by these models. Figure 3 presents the model results, including a comparison between the predicted and true proportions, as well as a comparison between the generated and original data using UMAP visualizations.

#### Fig.4 Gene correlation in original data vs generated data.
<img src="fig/Figure4_gene_correlation.png" width="50%" alt="Image description">
Fig.4 compares the correlational structure between the original data and the generated data.
-->
`_7_quick_demo` is a standalone directory that provides a quick glimpse of scRNA-seq data generation using pre-prepared data and models.

